app.R

```diff
@@ -20,6 +27,17 @@ ui <- fluidPage(
 )
 
 server <- function(input, output, session) {
+  # reactive expression to filter data based on user selections
+  filtered_data <- reactive({
+    data <- data |>
```

Comment:
I would assign this to something else, we don't want to overwrite our originally loaded dataframe (not sure if this does overwrite it but I wouldn't want to risk it)